src/averageAges.js

```diff
@@ -54,8 +68,23 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
-function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+function calculateAverageAgeDiff(people, onlyWithSon = false) {
+  const children = people.filter(child => {
+    return people.some(mother => child.mother === mother.name)
+      && (onlyWithSon
+        ? child.sex === 'm'
+        : true);
+  });
+
+  const ageDifferences = children
+    .map(child => child.born - people
+      .find(mother => mother.name === child.mother).born);
```

Comment:
```suggestion
    .map(child => child.born - people
      .find(mother => mother.name === child.mother)
      .born);
```
You can move the property to the same level as the method. It will read more clearly.